pipeline-runner/tests/testthat/test-gem2s-3-run_emptydrops.R

```diff
@@ -0,0 +1,35 @@
+mock_counts <- function() {
```

Comment:
an end-to-end test is missing that runs the whole gem2s pipeline with various inputs and various filters enabled and disabled.

Reply:
GEM2S steps can't be disabled (unlike QC steps). They are all run every time. I agree that we should add end-to-end test of gem2s pipeline but I think that this should be part of a seperate ticket as this one has already grown quite large.

Reply:
of course, I forgot that we are not able to disable steps, good point :) An end to end test as well as tests for the individual steps will definitely be super useful.

Reply:
our gem2s "end-to-end" test is the test_seurat_object function, which checks that the object has the correct slots. It doesn't however cross-check that the data is valid.

Reply:
why don't we move it to be where the other tests are?